crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -111,11 +117,20 @@ pub struct StaticFileProviderInner {
     metrics: Option<Arc<StaticFileProviderMetrics>>,
     /// Access rights of the provider.
     access: StaticFileAccess,
+    #[allow(dead_code)]
+    /// Write lock for when access is [StaticFileAccess::RW].
+    lock_file: Option<StorageLock>,
 }
 
 impl StaticFileProviderInner {
     /// Creates a new [`StaticFileProviderInner`].
     fn new(path: impl AsRef<Path>, access: StaticFileAccess) -> ProviderResult<Self> {
+        let lock_file = if access.is_read_write() {
+            Some(StorageLock::try_acquire(path.as_ref())?)
+        } else {
+            None
+        };
```

Comment:
should this not check the lock on read-only too?

Reply:
not acquiring, but checking if another process is using it as read-write? Process 1 acquires, process 2 checks and fails because process 1 is currently writing.